packages/worker/src/api/controllers/global/users.ts

```diff
@@ -22,6 +24,62 @@ export const save = async (ctx: any) => {
   }
 }
 
+export const bulkCreate = async (ctx: any) => {
+  let { users: newUsersRequested, groups } = ctx.request.body
+
+  if (!env.SELF_HOSTED && newUsersRequested.length > MAX_USERS_UPLOAD_LIMIT) {
+    ctx.throw(
+      400,
+      "Max limit for upload is 1000 users. Please reduce file size and try again."
```

Comment:
@shogunpurple @mike12345567 @Rory-Powell Checking for max length in these locations. this seem ok?

Reply:
@PClmnt - yeap blocking on the controller should work well.